gap/ExtraspecialNormalizerMatrixGroups.gi

```diff
@@ -500,61 +536,108 @@ function(m, q, type...)
-    generatingScalar := zeta ^ (QuoInt(q - 1, Gcd(q - 1, 2))) *
-    IdentityMat(2, F);
+    # We always need a generating scalar
+    if type = "L" then
+        generatingScalar := zeta ^ (QuoInt(q - 1, Gcd(q - 1, 2))) *
+                            IdentityMat(2 ^ m, F);
+    elif type = "S" then
+        generatingScalar := -IdentityMat(2 ^ m, F);
+    fi;
 
-    # Note that det(X1) = det(Y1) = 1, so we do not need to rescale these to
-    # determinant 1. Furthermore, det(V1) = 4 and this is always a square, so
-    # we can always rescale V1 to determinant 1.
-    scalarMultiplierV1 := ScalarToNormalizeDeterminant(V1, 2, F);
-    V1 := scalarMultiplierV1 * V1;
+    # Note that det(Xi) = det(Yi) = 1, so we do not need to rescale these to
+    # determinant 1. Similarly, det(Wi) = 1 for i >= 2.
+    #
+    # Furthermore, det(V1) = 2 ^ (2 ^ m) so we just have to rescale this by a
+    # factor of 1 / 2; this also leads to V1 preserving the symplectic form
+    # given by the block-diagonal matrix consisting of blocks [[0, 1], [-1, 0]]
+    listOfVi[1] := 1 / 2 * listOfVi[1];
 
+    # If type = "L", we have m = 1 and the only generator not considered yet is
+    # U1. Then det(U1) = 2 and we need to find a square root of 2 in GF(q).
+    #
+    # If type = "S", we still have to consider U1, the Vi for i >= 2 and W1.
+    # All these scale the symplectic form given by the block-diagonal matrix
+    # consisting of blocks [[0, 1], [-1, 0]] by a factor of 2 so we need to
+    # find a square root of 2 in GF(q) to fix this; this will also take care of
+    # the determinants.
     if IsEvenInt(e) or (p - 1) mod 8 = 0 or (p - 7) mod 8 = 0 then
-        # These are the cases where we can find a square root of det(U1) = 2 in
-        # GF(q) to rescale U1 to determinant 1.
-        scalarMultiplierU1 := ScalarToNormalizeDeterminant(U1, 2, F);
-        U1 := scalarMultiplierU1 * U1;
+        # These are the cases where we can find a square root of 2 (by quadratic
+        # reciprocity).
+        scalarMultiplier := 1 / RootFFE(F, 2 * zeta ^ 0, 2);
+        U1 := scalarMultiplier * U1;
+        listOfVi{[2..m]} := List(listOfVi{[2..m]}, Vi -> scalarMultiplier * Vi);
+        if m >= 2 then
+            listOfWi[1] := scalarMultiplier * listOfWi[1];
+        fi;
 
         generators := Concatenation([generatingScalar],
                                     generatorsOfNormalizerInGL.listOfXi,
                                     generatorsOfNormalizerInGL.listOfYi,
-                                    [U1, V1]);
+                                    [U1],
+                                    listOfVi,
+                                    listOfWi);
 
     else
-        # According to the Magma code, there is no need to take another
-        # generator instead of U1 if we cannot rescale it to determinant 1 - we
-        # simply discard U1 as a generator.
+        # Comparing with the Magma code, we can simply take 
```

Comment:
What do you mean here? Are you saying: "The Magma code does this, so we use this as justification for doing the same"? (that doesn't seem like a great argument, by the way ;-).

Or are you saying: "compared to the Magma code, we are better/different/whatever, and hence it suffices to do XYZ?

Reply:
The former :D ... well, sadly the relevant paper does not really say much (or rather anything) concerning this particular case so I trusted the Magma code on this one - however, doing this did pass some sanity checks: (1) the group sizes do come out the right way and (2) this case is exactly where an exception occurs in the structure of these C6 subgroups according to the tables in [BHR13] (namely, they are a bit smaller), so it makes sense that this should suffice.

One could of course check "theoretically" that these generators are enough, but I am not sure if this maybe transcends our current mathematical horizon a bit ;)